docs/contributor/standard_library.md

```diff
@@ -0,0 +1,28 @@
+# Standard Library
+
+Grain aims to have a comprehensive and consistent standard library.
+
+To us, this means that someone should be able to reach for the standard library to perform most of their day-to-day work, and it will always work the way they expect in the context of Grain.
+
+## Guidelines
+
+Here are some guidelines for making additions to the standard library while also keeping it high-quality and consistent!
+
+1. New data types should exist in their own modules. For example, the `Range` enum is the data type exported from `range.gr`. An exception to this are data types that are ubiquitous (`Option`/`Result`) or require compiler support (`List`).
```

Comment:
`List` doesn't actually get much for compiler support—I think it lives in Pervasives for the same reasons that Option and Result do. You just wanna be able to make them/annotate using them without having to import anything.
```suggestion
1. New data types should exist in their own modules. For example, the `Range` enum is the data type exported from `range.gr`. An exception to this are data types that are ubiquitous (`Option`/`Result`/`List`), which should live in `pervasives.gr`.
```